pkg/controller/replication/replication_controller_test.go

```diff
@@ -1116,3 +1128,189 @@ func BenchmarkGetPodControllerSingleNS(b *testing.B) {
+	validateSyncReplication(t, fakePodControl, 2, 0, 0)
+}
+
+func TestPatchPodWithOtherOwnerRef(t *testing.T) {
+	manager, fakePodControl := setupManagerWithGCEnabled()
+	rc := newReplicationController(2)
+	manager.rcStore.Indexer.Add(rc)
+	// add to podStore one more matching pod that doesn't have a controller
+	// ref, but has an owner ref pointing to other object. Expect a patch to
+	// take control of it.
+	unrelatedOwnerReference := api.OwnerReference{UID: util.NewUUID(), APIVersion: "batch/v1", Kind: "Job", Name: "Job"}
+	pod := newPod("pod", rc, api.PodRunning)
+	pod.OwnerReferences = []api.OwnerReference{unrelatedOwnerReference}
+	manager.podStore.Indexer.Add(pod)
+
+	err := manager.syncReplicationController(getKey(rc, t))
+	if err != nil {
+		t.Fatal(err)
+	}
+	// 1 patch to take control of pod, and 1 create of new pod.
+	validateSyncReplication(t, fakePodControl, 1, 0, 1)
+}
+
+func TestPatchPodWithCorrectOwnerRef(t *testing.T) {
+	manager, fakePodControl := setupManagerWithGCEnabled()
+	rc := newReplicationController(2)
+	manager.rcStore.Indexer.Add(rc)
+	// add to podStore a matching pod that has an ownerRef pointing to the rc,
+	// but ownerRef.Controller is false. Expect a patch to take control it.
+	rcOwnerReference := api.OwnerReference{UID: rc.UID, APIVersion: "v1", Kind: "ReplicationController", Name: rc.Name}
+	pod := newPod("pod", rc, api.PodRunning)
+	pod.OwnerReferences = []api.OwnerReference{rcOwnerReference}
+	manager.podStore.Indexer.Add(pod)
+
+	err := manager.syncReplicationController(getKey(rc, t))
+	if err != nil {
+		t.Fatal(err)
+	}
+	// 1 patch to take control of pod, and 1 create of new pod.
+	validateSyncReplication(t, fakePodControl, 1, 0, 1)
+}
+
+func TestPatchPodFails(t *testing.T) {
+	manager, fakePodControl := setupManagerWithGCEnabled()
+	rc := newReplicationController(2)
+	manager.rcStore.Indexer.Add(rc)
+	// add to podStore two matching pods. Expect two patches to take control
+	// them.
+	pod1 := newPod("pod1", rc, api.PodRunning)
+	pod2 := newPod("pod2", rc, api.PodRunning)
+	manager.podStore.Indexer.Add(pod1)
+	manager.podStore.Indexer.Add(pod2)
+	// let both patches fail. The rc manager will assume it fails to take
+	// control of the pods and create new ones.
+	fakePodControl.Err = fmt.Errorf("Fake Error")
+	err := manager.syncReplicationController(getKey(rc, t))
+	if err != nil {
+		t.Fatal(err)
+	}
+	// 2 patches to take control of pod1 and pod2 (both fail), 2 creates.
+	validateSyncReplication(t, fakePodControl, 2, 0, 2)
+}
+
+func TestPatchExtraPodsThenDelete(t *testing.T) {
+	manager, fakePodControl := setupManagerWithGCEnabled()
+	rc := newReplicationController(2)
+	manager.rcStore.Indexer.Add(rc)
+	// add to podStore three matching pods. Expect three patches to take control
+	// them, and later delete one of them.
+	pod1 := newPod("pod1", rc, api.PodRunning)
+	pod2 := newPod("pod2", rc, api.PodRunning)
+	pod3 := newPod("pod3", rc, api.PodRunning)
+	manager.podStore.Indexer.Add(pod1)
```

Comment:
optional nit: can create and add in one line if you don't reference the podX variables again.